assign-doi-datacite/src/main/java/no/unit/nva/datacite/clients/DoiClient.java

```diff
@@ -0,0 +1,81 @@
+package no.unit.nva.datacite.clients;
+
+import java.net.URI;
+import no.unit.nva.datacite.clients.exception.ClientException;
+import no.unit.nva.datacite.clients.models.Doi;
+
+/**
+ * NVAs supported DOI operations.
+ *
+ * <p>DOIs can exist in three states: draft, registered, and findable. DOIs are in the draft state when metadata
+ * have been registered, and will transition to the findable state when registering a landing page (URL).
+ *
+ * @see <a href="https://support.datacite.org/docs/landing-pages">Best practices for landing pages</a>
+ * @see <a href="https://support.datacite.org/docs/tombstone-pages">Best practices for Tombstone Landing pages</a>
+ * @see <a href="https://support.datacite.org/docs/versioning">Best practices about versioning</a>
+ */
+public interface DoiClient {
+
+    /**
+     * Create a DOI where registry agency (ie datacite) auto generates the suffix under the associated repository for
+     * NVA connected to the customer.
+     *
+     * @param customerId          NVAs customerId
+     * @param metadataDataciteXml datacite schema serialized xml as string
+     * @return {@link Doi} containing prefix/suffix ({@link Doi#toIdentifier()}) from provider
+     * @throws ClientException Error while communicating with Registry Agency
+     */
+    Doi createDoi(String customerId, String metadataDataciteXml) throws ClientException;
+
+    /**
+     * Update metadata for a given DOI.
+     *
+     * @param customerId          NVAs customerId
+     * @param metadataDataciteXml datacite schema serialized xml as string
```

Comment:
Currently it is not, because the Datacite MDS API accepts XML payloads directly. The Datacite REST API which eats other API payloads, has a field where they accept serialized base64 encoded XML payloads.